### ISSUE TITLE
Add min_doc_count option to geo grid aggregations

### ISSUE THREAD

**Issue by yutaka-nakano** [elastic/elasticsearch#50481]
Would it be possible to add `min_doc_count` option to geohash grid aggregation and geotile grid aggrigation?  
We use Elasticsearch and Kibana for geospatial data analysis. `min_doc_count` option would be very helpful for us to mask privacy information on dashboards with maps. It would also be helpful for excluding outliers.

We created a Pull Request for this feature. #50480 
Could you check it?

**Comment by gilesagnel** [comment]
Hi @wchaparro Can i pick this up?

**Comment by nreese** [comment]
There have been requests in Kibana for filtering geo aggregation buckets
* https://github.com/elastic/kibana/issues/149657
* https://github.com/elastic/kibana/issues/55520
* https://github.com/elastic/kibana/issues/63401

**Comment by timDINOS** [comment]
Hi @yutaka-nakano, is this still open? I would like to fix this issue.

**Comment by iverase** [comment]
@timDINOS yes it is still open and I think nobody is actively working on that so please feel free to work on it.

**Comment by gapspaceengineers** [comment]
So:

1. Check the files changed in the [PR request](https://github.com/elastic/elasticsearch/pull/50480/commits/86d2c629f8382f6caca72d255158b7b9349e4a13#diff-ad77925c01b996746e9f50f86d644d5ebe6460ea9ba12d286ce765e86442c694)

2.

**Comment by gapspaceengineers** [comment]
So, would this work?

1. Check the files that were changed in the [PR request ](https://github.com/elastic/elasticsearch/pull/50480/commits/86d2c629f8382f6caca72d255158b7b9349e4a13#diff-ad77925c01b996746e9f50f86d644d5ebe6460ea9ba12d286ce765e86442c694) mentioned in OPs post.

2. Look into those changes and apply same the functionality (and tests I guess) in spirit of the above PR for the following cases:
org/elasticsearch/search/aggregations/bucket/geogrid/**GeoHashGridAggregator.java**
org/elasticsearch/search/aggregations/bucket/geogrid/**GeoTileGridAggregator.java**
org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/**GeoHexGridAggregator.java**

Keeping in mind to exclude support for _min_doc_count=0_ (like in the PR as the default is 1)

**Comment by Kiriakos1998** [comment]
Hello, @timothy-choi  are you working on this issue? I am looking for my first issue to attempt my first contribution and If no one is working with this one I would love to give it a shot.
